### DISCUSSION TITLE
Objects with issues

### DISCUSSION THREAD

**Discussion by luismrguimaraes** [charlesneimog/pd4web#22] [Q&A 🇺🇸]
- [x] [else/comb.rev~]
- [x] [else/compress~]

Patch:
```Pd
#N canvas 160 -1109 899 487 12;
#N canvas 460 171 450 385 subpatch 0;
#X obj 111 56 inlet~;
#X obj 126 251 outlet~;
#X obj 120 111 else/comb.rev~ 50 0.3 0.3 0.3;
#X floatatom 234 204 7 0 0 0 - - - 0;
#X obj 121 166 else/compress~;
#X connect 0 0 2 0;
#X connect 2 0 4 0;
#X connect 4 0 1 0;
#X connect 4 1 3 0;
#X restore 298 128 pd subpatch;
#X obj 300 235 dac~;
#X obj 313 54 osc~ 440;
#X obj 329 98 *~;
#X obj 297 195 *~ 0.15;
#X obj 369 67 else/pulse~ 1 0.25;
#X connect 0 0 4 0;
#X connect 2 0 3 0;
#X connect 3 0 0 0;
#X connect 4 0 1 0;
#X connect 4 0 1 1;
#X connect 5 0 3 1;

```
- [x] [else/plate.rev~] and [else/allpass.rev~]:

Patch:
```Pd
#N canvas 324 361 899 487 12;
#N canvas 520 509 838 404 plateVerb 0;
#X obj 71 15 inlet~;
#X obj 111 194 outlet~;
#X obj 152 11 inlet;
#X obj 125 112 else/plate.rev~;
#X obj 200 195 outlet~;
#X obj 203 13 loadbang;
#X msg 230 46 damp 0.25;
#X msg 128 51 wet \$1;
#X msg 250 72 size 0.01;
#X connect 0 0 3 0;
#X connect 2 0 7 0;
#X connect 3 0 1 0;
#X connect 3 1 4 0;
#X connect 5 0 6 0;
#X connect 6 0 3 0;
#X connect 7 0 3 0;
#X connect 8 0 3 0;
#X restore 306 21 pd plateVerb;
```

- [ ] [else/sample~] - doesn't find audio files.

**Comment by charlesneimog** [comment]
@luismrguimaraes, `[else/comb.rev~]` and `[else/compress~]` are working for me and in my CI tests.

**Reply by luismrguimaraes** [reply]
Have you tried the patch I shared?
This one:
```
#N canvas 160 -1109 899 487 12;
#N canvas 460 171 450 385 subpatch 0;
#X obj 111 56 inlet~;
#X obj 126 251 outlet~;
#X obj 120 111 else/comb.rev~ 50 0.3 0.3 0.3;
#X floatatom 234 204 7 0 0 0 - - - 0;
#X obj 121 166 else/compress~;
#X connect 0 0 2 0;
#X connect 2 0 4 0;
#X connect 4 0 1 0;
#X connect 4 1 3 0;
#X restore 298 128 pd subpatch;
#X obj 300 235 dac~;
#X obj 313 54 osc~ 440;
#X obj 329 98 *~;
#X obj 297 195 *~ 0.15;
#X obj 369 67 else/pulse~ 1 0.25;
#X connect 0 0 4 0;
#X connect 2 0 3 0;
#X connect 3 0 0 0;
#X connect 4 0 1 0;
#X connect 4 0 1 1;
#X connect 5 0 3 1;
```

They compile but I get "error: ... couldn't create" in the browser.

**Reply by charlesneimog** [reply]
Yes, I can compile the patch here.

[WebPatch.zip](https://github.com/user-attachments/files/17717114/WebPatch.zip)

**Reply by luismrguimaraes** [reply]
That patch works here as well... 

I'm using version `2.1.0.dev2` of pd4web (which results in the errors). Is there anything else I can try to delete besides `build/` and `Pd4Web/`?
Also, something I noticed is that while your .data file is readable in a text editor, mine isn't.

Here's the patch:
[WebPatch.zip](https://github.com/user-attachments/files/17718159/WebPatch.zip)

**Reply by charlesneimog** [reply]
Try to delete also the WebPatch.

The .data file is all the files from your project, to build the filesystem used inside the Worklet. So maybe you are using some audio files or binary files then it will not be recognized as text.

Have you tried to compile the patch from a clean folder?

**Reply by luismrguimaraes** [reply]
[WebPatch.zip](https://github.com/user-attachments/files/17719078/WebPatch.zip)

I've just tried in a clean folder and without any of the other folders (Libs/ and Audios/) and even though the .data is readable, the patch still throws the same errors.

Should I also delete the contents in `Users/<user>/Library/pd4web/` and try again?

**Reply by charlesneimog** [reply]
Yes, you can try running `pip uninstall pd4web` and then reinstalling it directly from the Git repository.

Before reinstalling, make sure that the `pd4web` executable is completely removed. I've noticed that sometimes in a conda environment, if the installation wasn't clean, `pip` might not update the `pd4web` executable properly. So double-check that it’s no longer present before proceeding with the new installation.

**Reply by luismrguimaraes** [reply]
It's working now, but using the flag `-nogui` breaks the code, because `pd4web.js` cannot run `setSoundIcon`. Without the flag, I get Pd running but get the error `Pd4WebAudioSwitch not found`.
